tests/embed/test_splitae.py

```diff
@@ -0,0 +1,62 @@
+from multiview.embed.splitae import SplitAE
+import numpy as np
+import sklearn.cluster
+import sklearn.datasets
+from itertools import permutations
+import torch
+
+# These are blobs that PCA can reduce to nice clusters easily, so splitAE should be able to aswell
+# cluster_std is high s.t. any 2 features alone do not make nice seperable blobs
+def test_splitae_blobs():
+    accuracies = []
+    for i in range(10):
+        view1, labels = sklearn.datasets.make_blobs(n_samples=1000, n_features=20, centers=None, cluster_std=10.0, center_box=(-10.0, 10.0), shuffle=True, random_state=None)
+        # plt.scatter(*view1.T[2:4], c=labels)
+        linear = torch.nn.Linear(20, 20)
+        view2 = linear(torch.FloatTensor(view1)).detach().cpu() # view2 is just linear transform of view1
+        trainSplit = 800
+
+        # 0 hidden layers -- i.e. no nonlinearities, just matrix mults -- can solve this problem
+        splitae = SplitAE(hiddenSize=64, numHiddenLayers=0, embedSize=2, trainingEpochs=10, batchSize=10, learningRate=0.01, printInfo=False, printGraph=False)
+        splitae.fit([view1[:trainSplit], view2[:trainSplit]], validationXs=[view1[trainSplit:], view2[trainSplit:]])
+
+        embeddings, reconstructedView1, predictedView2 = splitae.transform([view1[-200:]])
+        # plt.scatter(*embeddings.T, c=labels[trainSplit:])
+        kmeans = sklearn.cluster.KMeans(n_clusters=3).fit(embeddings)
+        # plt.scatter(*embeddings.T, c=(kmeans.labels_))
+
+        def clusterAccuracy(prediction, target, nClusters):
+            predictionPermuted = prediction.copy()
+            clusterPermutations = list(permutations(range(nClusters)))
+            maxAccuracy = 0
+            for permutation in clusterPermutations:
+                indexes = []
+                for i in range(nClusters):
+                    indexes.append(prediction == i)
+                for i in range(nClusters):
+                    predictionPermuted[indexes[i]] = permutation[i]
+                accuracy = np.sum(predictionPermuted == target) / len(target)
+                maxAccuracy = accuracy if accuracy > maxAccuracy else maxAccuracy
+            return maxAccuracy
+
+        accuracy = clusterAccuracy(kmeans.labels_, labels[trainSplit:], nClusters=3)
+        accuracies.append(accuracy)
+    # distribution of accuracies sometimes has outliers around 0.65, so be conservative
+    assert np.mean(accuracies) > 0.7
```

Comment:
Rather than doing this, can you set the seed for the make_blobs function so that you can know exactly how well the algorithm will perform, then assert that the accuracies are within a numerical tolerance of what you expect? Because if in the future you change something else in the algorithm that somehow messes up its ability to predict as well, but it still passes 0.7, it'll pass this test even though its predictive ability changed. Let me know what you think

Reply:
Hmmm, I see what you're saying -- that maybe setting the make blobs seed will allow me to set a stricter bound like 0.9 or so. I think the variability in accuracy is due both the variability in make blobs as well as variability in convergence of the model (weights are randomly initialized). But good idea I'll do a test and see if this is the case

Reply:
So it looks like if I add the seed I can increase the bound a little bit to around 0.8  . But I think the added benefit of testing on different random make_blobs cases (some of which might expose a new weakness if the code is changed detrimentally) outweighs that increase since it's pretty small.

Reply:
i don't know about changing the bound, however, random seeds for tests like this are pretty essential, otherwise you could be chasing failed builds because of nothing other than poor performance on something random

Reply:
Maybe we should save the data if the test fails because a failure on random data might reveal an edge case where the algorithm performs especially poorly. Or maybe that might just reveal data which is inherently impossible to cluster? Not sure.